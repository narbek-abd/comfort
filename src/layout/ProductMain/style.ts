import styled from "styled-components";

export const ProductMain = styled.div`


`

export const Inner = styled.div`


`
 