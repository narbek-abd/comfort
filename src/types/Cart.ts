interface productItem {
    id: number;
    quantity: number
}

export interface CartState {
    products: productItem[] | []
}


export enum CartActionTypes {
    ADD_PRODUCTS = 'ADD_PRODUCTS',
    REMOVE_PRODUCT = 'REMOVE_PRODUCT',
    CHANGE_PRODUCT_COUNT = 'CHANGE_PRODUCT_COUNT',
}


interface AddProductsAction {
    type: CartActionTypes.ADD_PRODUCTS;
    payload: any
}
interface RemoveProductsAction {
    type: CartActionTypes.REMOVE_PRODUCT;
    payload: any;
}

interface ChangeProductCountAction  {
    type: CartActionTypes.CHANGE_PRODUCT_COUNT;
    payload: any;
}


export type CartAction = AddProductsAction | RemoveProductsAction | ChangeProductCountAction
