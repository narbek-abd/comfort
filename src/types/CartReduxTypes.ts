export interface ProductItemTypes {
    id: number;
    quantity: number;
}

export interface CartState {
    products: ProductItemTypes[] | [];
}

export enum CartActionTypes {
    ADD_PRODUCTS = "ADD_PRODUCTS",
    REMOVE_PRODUCT = "REMOVE_PRODUCT",
    CHANGE_PRODUCT_COUNT = "CHANGE_PRODUCT_COUNT",
    CLEAR_CART = "CLEAR_CART",
}

interface AddProductsAction {
    type: CartActionTypes.ADD_PRODUCTS;
    payload: ProductItemTypes[];
}
interface RemoveProductsAction {
    type: CartActionTypes.REMOVE_PRODUCT;
    payload: ProductItemTypes[];
}

interface ChangeProductCountAction {
    type: CartActionTypes.CHANGE_PRODUCT_COUNT;
    payload: ProductItemTypes[];
}

interface ClearCart {
    type: CartActionTypes.CLEAR_CART;
}

export type CartAction =
    | AddProductsAction
    | RemoveProductsAction
    | ChangeProductCountAction
    | ClearCart;
