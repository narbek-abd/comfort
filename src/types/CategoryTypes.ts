export interface CategoryTypes {
	id: number;
	name: string;
	parent_id: number;
	children?: CategoryTypes[];
}