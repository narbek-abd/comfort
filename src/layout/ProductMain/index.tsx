import React from 'react';
import * as G from "../../globalStyle";

import * as S from "./style";

interface ProductMainProps {
  link: any;
  hasChildren: boolean;
}

const ProductMain: React.FC<ProductMainProps> = () => {
  return (
    <S.ProductMain>
      <G.Container>
        <S.Inner>
        
        </S.Inner>
      </G.Container>
      
    </S.ProductMain>
  );
};

export default ProductMain;