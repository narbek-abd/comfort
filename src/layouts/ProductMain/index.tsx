import React from "react";
import * as G from "../../globalStyle";
import ProductGallery from "../../components/ProductGallery";

import * as S from "./style";
import { Icon } from "../../components/Icon";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from '../../store/redusers';
import { addProduct } from '../../store/redusers/CartReduser';

interface ProductMainProps {
  product: any;
}

const ProductMain = ({ product }: ProductMainProps) => {
  const dispatch = useDispatch();

  function addToCart() {
    dispatch(addProduct(product))
  }

  return (
    <S.ProductMain>
      <G.Container>
        <S.Inner>
          <S.Left>
            <ProductGallery images={images} />
          </S.Left>
          <S.Right>
            <S.Name>{product.title}</S.Name>

            <S.Rating>
              <S.Stars>⭐️⭐️⭐️⭐️⭐️</S.Stars>

              <span>({product.stock})</span>
            </S.Rating>

            <S.Price>
              {product.price}
              <span>$40.00</span>
            </S.Price>

            <S.Color>
              <p>Color</p>

              <div>
                <span></span>
              </div>
            </S.Color>

            <S.Desc>{product.description}</S.Desc>

            <S.Actions>
              <G.Button onClick={addToCart}>Add To Cart</G.Button>

              <S.Like>
                <Icon name="heart" />
              </S.Like>
            </S.Actions>

            <S.Categories>
              <p>Categories:</p>
              <div>
                <Link to="/">Chairs</Link>
              </div>
            </S.Categories>

            <S.Tags>
              <p>Tags:</p>
              <div>
                <Link to="/">Chairs</Link>
              </div>
            </S.Tags>
          </S.Right>
        </S.Inner>
      </G.Container>
    </S.ProductMain>
  );
};

const images = [
  {
    id: 1,
    url: "https://images.pexels.com/photos/7863536/pexels-photo-7863536.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940",
    main: true,
  },
  {
    id: 2,
    url: "https://images.pexels.com/photos/3771691/pexels-photo-3771691.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940",
  },
  {
    id: 3,
    url: "https://images.pexels.com/photos/2258083/pexels-photo-2258083.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940",
  },
  {
    id: 4,
    url: "https://images.pexels.com/photos/7606067/pexels-photo-7606067.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940",
  },
  {
    id: 5,
    url: "https://images.pexels.com/photos/8003996/pexels-photo-8003996.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940",
  },
];

export default ProductMain;
