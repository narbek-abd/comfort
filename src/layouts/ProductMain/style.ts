import styled from "styled-components";
import { Container } from "../../globalStyle";
import { Button } from '../../components/Button/style';

export const ProductMain = styled.section`
	margin-top: 150px;

	@media (max-width: 992px) {
		${Container} {
			padding: 0;
			max-width: 1500px;
		}
	}
`;

export const Inner = styled.div`
	display: flex;

	@media (max-width: 992px) {
		display: block;
	}
`;

export const Left = styled.div`
	width: 547px;
	@media (max-width: 992px) {
		width: auto;
	}
`;

export const Right = styled.div`
	color: var(--color-dark-blue);
	font-family: Josefin Sans;
	font-weight: 600;
	margin-left: 44px;
`;

export const Name = styled.h3`
	font-size: 36px;
	line-height: 36px;
	color: #0d134e;
	margin-top: 74px;
`;

export const Rating = styled.div`
	margin-top: 13px;
	font-family: Josefin Sans;
	font-weight: 100;
	font-size: 14px;
	line-height: 29px;
	display: flex;
`;

export const Stars = styled.div`
	margin-right: 5px;
`;

export const Price = styled.div`
	font-family: Josefin Sans;
	font-weight: 100;
	font-size: 16px;
	line-height: 29px;
	color: var(--color-dark-blue);

	span {
		margin-left: 32px;
		text-decoration-line: line-through;
		color: #fb2e86;
	}
`;

export const Color = styled.div`
	margin-top: 12px;
	line-height: 16px;
	span {
		margin-right: 8px;
	}
`;

export const Desc = styled.p`
	margin-top: 12px;
	font-size: 16px;
	line-height: 29px;
	color: #a9acc6;
`;

export const Actions = styled.div`
	margin-top: 34px;
	display: flex;
	align-items: center;
`;

export const Like = styled.div`
	cursor: pointer;
	${Button} {
	padding: 17px 25px;
		background-color: transparent;
		border: none;
	}
`;

export const Categories = styled.div`
	margin-top: 17px;
	display: flex;

	div {
		margin-left: 16px;
	}
`;
