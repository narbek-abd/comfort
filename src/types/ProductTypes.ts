import { CategoryTypes } from "./CategoryTypes";
import { ImageTypes } from "./ImageTypes";

export interface ProductTypes {
    id: number;
    name: string;
    price: number;
    quantity: number;
    category_id: number;
    description: string;
    category: CategoryTypes;
    images: ImageTypes[];
    storageQuantity?: number;
}