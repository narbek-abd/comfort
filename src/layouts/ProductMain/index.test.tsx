import { screen } from "@testing-library/react";
import { render, store } from "../../test/test-utils";
import "../../test/mock";
import ProductMain from "./index";
import axios from "axios";
import { ProductTypes } from "../../types/ProductTypes";

describe("ProductMain component", () => {
	let product: ProductTypes;

	beforeAll(() => {
		product = {
			id: 1,
			name: "Iphone",
			price: 123,
			quantity: 12,
			category_id: 1,
			description: '',
			category: {
				id: 1,
				slug: 'phones',
				parent_id: 2,
				name: "phones",
			},
			images: [{ id: 1, image: "https://image.png" }],
		};
	});

	test("should display product data", () => {
		const { container } = render(<ProductMain product={product} />);

		expect(container).toMatchSnapshot();
	});
});
