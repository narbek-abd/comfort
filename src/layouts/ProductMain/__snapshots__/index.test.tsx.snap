// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductMain component should display product data 1`] = `
<div>
  <section
    class="sc-egiyK lgVpO"
  >
    <div
      class="sc-bdvvtL gFjKpS"
    >
      <div
        class="sc-bqiRlB iRNXfC"
      >
        <div
          class="sc-ksdxgE hmrQvX"
        >
          <div
            class="sc-furwcr evUCzD"
          >
            <div
              class="sc-pVTFL kgiFwU"
            >
              <div
                class="swiper-button-prev"
              >
                <svg
                  class="icon icon-arrow"
                >
                  <use
                    xlink:href="symbol-defs.svg#arrow"
                  />
                </svg>
              </div>
              <div
                class="swiper swiper-initialized swiper-horizontal swiper-pointer-events"
                style="height: 490px;"
              >
                <div
                  class="swiper-wrapper"
                  style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
                >
                  <div
                    class="swiper-slide"
                    data-image-id="1"
                  >
                    <div
                      class="sc-jrQzAO dULXpj"
                    >
                      <img
                        alt="product img"
                        data-testid="1"
                        src="http://localhost:8000/storage/https://image.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="swiper-button-next"
              >
                <svg
                  class="icon icon-arrow"
                >
                  <use
                    xlink:href="symbol-defs.svg#arrow"
                  />
                </svg>
              </div>
            </div>
            <div
              class="sc-kDTinF dzDmCB"
            >
              <div
                class="sc-iCfMLu cFiRNz"
              >
                <img
                  data-testid="main-1"
                  src="http://localhost:8000/storage/https://image.png"
                  style="transform: translate(-50%, -50%) scale(1);"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-hBUSln lljyfE"
        >
          <h3
            class="sc-fotOHu fuObcD"
          >
            Iphone
          </h3>
          <div
            class="sc-fFeiMQ dQcjYt"
          >
            <div
              class="sc-bkkeKt dLdkR"
            >
              ⭐️⭐️⭐️⭐️⭐️
            </div>
            <span>
              (55)
            </span>
          </div>
          <div
            class="sc-ieecCq jJsfOQ"
          >
            123
            <span>
              $40.00
            </span>
          </div>
          <div
            class="sc-dJjYzT hpkcdk"
          >
            <span>
              Quantity: 
            </span>
            <span>
              12
            </span>
          </div>
          <p
            class="sc-hGPBjI cERqgi"
          />
          <div
            class="sc-dlVxhl foLraq"
          >
            <button
              class="sc-crHmcD vGuTY"
            >
              Add To Cart
            </button>
            <div
              class="sc-kfPuZi kwcZcE"
            >
              <button
                class="sc-crHmcD cdydW"
                color="blue"
              >
                <svg
                  class="icon icon-heart"
                >
                  <use
                    xlink:href="symbol-defs.svg#heart"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="sc-fKVqWL hksQPY"
          >
            <p>
              Category:
            </p>
            <div>
              <a
                href="/catalog/phones"
              >
                phones
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
